### PR TITLE
render: switch to generic render from library-go

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: cmd/cluster-kube-apiserver-operator/render/render.go [hunk 1]
```diff
@@ -1,57 +1,48 @@
...
+		manifest: *genericrenderoptions.NewManifestOptions("kube-apiserver", "openshift/origin-hypershift:latest"),
+
+		lockHostPath:   "/var/run/kubernetes/lock",
+		etcdServerURLs: []string{"https://127.0.0.1:2379"},
+		etcdServingCA:  "root-ca.crt",
+	}
 	cmd := &cobra.Command{
 		Use:   "render",
-		Short: "Render kubernetes API server bootstrap manifests, secrets and configMaps",
+		Short: "Render kubernetes controller manager bootstrap manifests, secrets and configMaps",
```
Comment (tnozicka) [inline cmd/cluster-kube-apiserver-operator/render/render.go]: typo

Comment (sttts) [reply cmd/cluster-kube-apiserver-operator/render/render.go]: fixed

File: cmd/cluster-kube-apiserver-operator/render/render.go [hunk 2]
```diff
@@ -62,176 +53,98 @@ func NewRenderCommand() *cobra.Command {
...
-	bootstrapOverrides, err := readFileTemplate(tlsOverride, data)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load config override file %q: %v", tlsOverride, err)
-	}
-	// TODO: Remove this when the flag is gone
-	if len(r.deprecatedConfigOverrideFile) > 0 {
-		r.configOverrideFiles = append(r.configOverrideFiles, r.deprecatedConfigOverrideFile)
+// Run contains the logic of the render command.
+func (r *renderOpts) Run() error {
+	if err := r.Complete(); err != nil {
```
Comment (soltysh) [inline cmd/cluster-kube-apiserver-operator/render/render.go]: That's unusual and definitely not in-line with all of our CLI. We usually call the Validate-Complete-Run triad inside of cobra's Run, I think that should be true here as well.

File: cmd/cluster-kube-apiserver-operator/render/render.go [hunk 3]
```diff
@@ -62,176 +53,98 @@ func NewRenderCommand() *cobra.Command {
...
-		return nil, fmt.Errorf("failed to merge configs: %v", err)
+	if err := r.manifest.ApplyTo(&renderConfig.ManifestConfig); err != nil {
+		return err
 	}
-	yml, err := yaml.JSONToYAML(mergedConfig)
-	if err != nil {
-		return nil, err
+	if err := r.generic.ApplyTo(
+		&renderConfig.FileConfig,
+		genericrenderoptions.Template{FileName: "defaultconfig.yaml", Content: v311_00_assets.MustAsset(filepath.Join(bootstrapVersion, "kube-apiserver", "defaultconfig.yaml"))},
```
Comment (tnozicka) [inline cmd/cluster-kube-apiserver-operator/render/render.go]: make the filenames const?

File: cmd/cluster-kube-apiserver-operator/render/render.go (hunk 2 above)
Comment (sttts) [reply cmd/cluster-kube-apiserver-operator/render/render.go]: done

File: cmd/cluster-kube-apiserver-operator/render/render.go [hunk 4]
```diff
@@ -62,176 +53,98 @@ func NewRenderCommand() *cobra.Command {
...
 	return cmd
 }
 
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	r.manifest.AddFlags(fs, "apiserver")
+	r.generic.AddFlags(fs, kubecontrolplanev1.GroupVersion.WithKind("KubeAPIServerConfig"))
+
+	fs.StringVar(&r.lockHostPath, "manifest-lock-host-path", r.lockHostPath, "A host path mounted into the apiserver pods to hold lock.")
+	fs.StringArrayVar(&r.etcdServerURLs, "manifest-etcd-server-urls", r.etcdServerURLs, "The etcd server URL, comma separated.")
+	fs.StringVar(&r.etcdServingCA, "manifest-etcd-serving-ca", r.etcdServingCA, "The etcd serving ca.")
```
[the remark this answers is not in the document]
Comment (sttts) [reply cmd/cluster-kube-apiserver-operator/render/render.go]: done

**Pull request merged**